File: week07/snippets/prepare_gtfs.mjs
import {parse, stringify} from 'csv/sync';
import fs from 'node:fs/promises'

const __dirname = new URL('.', import.meta.url).pathname;
const RAW_DATA_DIR = __dirname + 'raw_data/';
const PROCESSED_DATA_DIR = __dirname + 'processed_data/';

const GTFS_FEEDS = ['septa_bus', 'septa_rail'];

for (const gtfsFeed of GTFS_FEEDS) {
  const gtfsFeedFolder = RAW_DATA_DIR + gtfsFeed + '/';
  for (const gtfsFileName of await fs.readdir(gtfsFeedFolder)) {

    // Read the data from the raw GTFS CSV file.
    const gtfsFilePath = gtfsFeedFolder + gtfsFileName;
    const content = await fs.readFile(gtfsFilePath, 'utf8');
    const data = parse(content, {skip_empty_lines: true});

    // Write the data to a new processed CSV file, creating the folder if it
    // doesn't exist.
    const outputFolder = PROCESSED_DATA_DIR + gtfsFeed + '/';
    const outputPath = outputFolder + gtfsFileName.replace('.txt', '.csv');
    await fs.mkdir(outputFolder, {recursive: true});
    await fs.writeFile(outputPath, stringify(data), 'utf8');
  }
}

File: week07/snippets/prepare_gtfs.py
import csv
import pathlib

RAW_DATA_DIR = pathlib.Path(__file__).parent / 'raw_data'
PROCESSED_DATA_DIR = pathlib.Path(__file__).parent / 'processed_data'

GTFS_FEEDS = ['septa_bus', 'septa_rail']

for gtfs_feed in GTFS_FEEDS:
    gtfs_feed_folder = RAW_DATA_DIR / gtfs_feed
    for gtfs_path in gtfs_feed_folder.iterdir():

        # Read the data from the raw GTFS CSV file.
        with gtfs_path.open('r', encoding='utf-8') as f:
            reader = csv.reader(f)
            data = [row for row in reader]

        # Write the data to a new processed CSV file, creating the
        # necessary directories if they don't exist.
        output_folder = PROCESSED_DATA_DIR / gtfs_feed
        output_path = output_folder / f'{gtfs_path.stem}.csv'
        output_folder.mkdir(parents=True, exist_ok=True)
        with output_path.open('w', encoding='utf-8') as f:
            writer = csv.writer(f)
            writer.writerows(data)
